stempy/image.cpp

```diff
@@ -142,14 +142,19 @@ void _runCalculateSTEMValues(const uint16_t data[],
     vtkm::cont::make_ArrayHandle(data, numberOfPixels * imageNumbers.size());
 #endif
   for (unsigned i = 0; i < imageNumbers.size(); ++i) {
+    // We need to ensure we are using uint64_t to prevent overflow.
+    auto offset = static_cast<uint64_t>(i) * numberOfPixels;
 #ifdef VTKm
     // Use view to the array already transfered
-    auto view = vtkm::cont::make_ArrayHandleView(dataHandle, i * numberOfPixels,
+    // Note: We need to ensure the offset is pass as a uint64 to prevent
+    // overflow.
+    auto view = vtkm::cont::make_ArrayHandleView(dataHandle, offset,
```

Comment:
It looks like the second argument to the `make_ArrayHandleView` function is a `vtkm::Id`, and this is defined as an `int64_t` as long as the cmake option `VTKm_USE_64BIT_IDS` is true (it is ON by default).

I hope `int64_t` is big enough (since `offset` is a `uint64_t`). As long as it is big enough, I think this should be good.

Note that if `VTKm_USE_64BIT_IDS` is off, `vtkm::Id` is `int32_t` instead.

Reply:
Yes, we should be good.